api/pkg/cli/cmd/install/install.go

```diff
@@ -169,55 +177,73 @@ func (opts *options) isResourceNotFoundError(err error) error {
+						strings.Title(opts.resource.GetKind()), opts.resource.GetName(), existingVersion, opts.cs.Namespace())
+
+				case exVer.LessThan(newVer):
+					if opts.version == "" {
+						resourceVersion = resourceVersion + "(latest)"
+					}
+					return fmt.Errorf("%s %s(%s) already exists in %s namespace. Use upgrade command to install v%s",
+						strings.Title(opts.resource.GetKind()), opts.resource.GetName(), existingVersion, opts.cs.Namespace(), resourceVersion)
+
+				case exVer.GreaterThan(newVer):
+					return fmt.Errorf("%s %s(%s) already exists in %s namespace. Use downgrade command to install v%s",
+						strings.Title(opts.resource.GetKind()), opts.resource.GetName(), existingVersion, opts.cs.Namespace(), resourceVersion)
 
-			case exVer.GreaterThan(newVer):
-				return fmt.Errorf("%s %s(%s) already exists in %s namespace. Use downgrade command to install v%s",
-					strings.Title(opts.resource.GetKind()), opts.resource.GetName(), existingVersion, opts.cs.Namespace(), newVersion)
+				default:
+					return fmt.Errorf("%s %s(%s) already exists in %s namespace. Use reinstall command to overwrite existing",
+						strings.Title(opts.resource.GetKind()), opts.resource.GetName(), existingVersion, opts.cs.Namespace())
+				}
 
-			default:
-				return fmt.Errorf("%s %s(%s) already exists in %s namespace. Use reinstall command to overwrite existing",
-					strings.Title(opts.resource.GetKind()), opts.resource.GetName(), existingVersion, opts.cs.Namespace())
+			} else {
+				return fmt.Errorf("%s %s already exists in %s namespace but seems to be missing version label. Use reinstall command to overwrite existing",
+					strings.Title(opts.resource.GetKind()), opts.resource.GetName(), opts.cs.Namespace())
 			}
+		}
 
-		} else {
-			return fmt.Errorf("%s %s already exists in %s namespace but seems to be missing version label. Use reinstall command to overwrite existing",
-				strings.Title(opts.resource.GetKind()), opts.resource.GetName(), opts.cs.Namespace())
+		if err == installer.ErrVersionIncompatible {
+			return fmt.Errorf("Task %s(%s) requires Tekton Pipelines min version v%s but found %s", opts.name(), resourceVersion, resourcePipelineMinVersion, pipelinesVersion)
 		}
-	}
 
-	if strings.Contains(err.Error(), "mutation failed: cannot decode incoming new object") {
-		version, vErr := opts.hubRes.MinPipelinesVersion()
-		if vErr != nil {
-			return vErr
+		if strings.Contains(err.Error(), "mutation failed: cannot decode incoming new object") {
+			return fmt.Errorf("%v \nMake sure the pipeline version you are running is not lesser than %s and %s have correct spec fields",
+				err, resourcePipelineMinVersion, opts.kind)
 		}
-		return fmt.Errorf("%v \nMake sure the pipeline version you are running is not lesser than %s and %s have correct spec fields",
-			err, version, opts.kind)
 	}
-	return err
+
+	if errors[0] != installer.ErrWarnVersionNotFound {
```

Comment:
are you sure this wil be present at only first location in all the cases if it happens

Reply:
yeah...the sequence of storing errors is such that first it will check for the Pipelines version from the pipelines-controller pod and if not found then it will push this error into the errors array.